src/rngs/os.rs

```diff
@@ -1102,6 +1102,76 @@ mod imp {
     }
 }
 
+#[cfg(all(target_arch = "wasm32",
+          not(target_os = "emscripten"),
+          not(feature = "stdweb"),
+          feature = "wasm-bindgen"))]
+mod imp {
+    use core::fmt;
+
+    use __wbg_shims::*;
+
+    use {Error, ErrorKind};
+    use super::OsRngImpl;
+
+    #[derive(Clone)]
+    pub enum OsRng {
+        Node(NodeCrypto),
+        Browser(BrowserCrypto),
+    }
+
+    impl OsRngImpl for OsRng {
+        fn new() -> Result<OsRng, Error> {
+            if this.window().is_undefined() {
+                return Ok(OsRng::Node(node_require("crypto")))
```

Comment:
Is the logic inverted here or is it simply assumed that `node_require("crypto")` will not fail?

I *assume* that `window()` being undefined implies we can't use browser-crypto, so I guess this is just missing a check that `NodeCrypto` is available?

(If it really cannot fail then I don't understand why you return errors below instead of falling back to `Node`.)

Reply:
Currently it's assumed that `require("crypto")` won't fail as I figured that node has had that API for quite some time. Is that something that should be checked for?

I figured below it'd be good to test that if we see we're in a browser whether crypto actually works, and falling back to an error if it looks like an older browser or something like that.

Reply:
The `crypto` module has been around since Node 0.x so it's safe to assume importing it won't fail.

`crypto.randomFillSync` was backported to Node 6.x, the oldest version still being maintained, so that's also safe to use blindly.

Reply:
Thanks! I've updated with a comment about this